templates/pages/reset-password.jsx

```diff
@@ -30,22 +41,79 @@ var ResetPassword = React.createClass({
                 className="emailSent centerDiv"
                 headerClass="emailSentHeader"
                 header="Check your email">
-                  <p>We've emailed you instructions for creating a new password.</p>
+                  <p>We&rsquo;ve emailed you instructions for creating a new password.</p>
               </IconText> : false}
             {this.state.email ?
-              <ResetView username="" submitForm={this.handleRequestPassword}/> : false
+              <ResetView username={this.state.queryObj.uid} submitForm={this.handleRequestPassword}/> : false
             }
         </div>
       </div>
     );
   },
   handleRequestPassword: function(error, data) {
-    console.log("inside App we see:", error, data);
+    if ( error ) {
+      console.error("validation error", error);
+      return;
+    }
+
+    fetch('/reset-password', {
+      method: 'post',
+      headers: {
+        'Accept': 'application/json; charset=utf-8',
+        'Content-Type': 'application/json; charset=utf-8'
+      },
+      body: JSON.stringify({
+        uid: data.username,
+        password: data.password,
+        resetCode: this.state.queryObj.resetCode
+      })
+    }).then(function(response) {
+      var queryObj,
+        redirectObj;
+      if ( response.status === 200 ) {
```

Comment:
is there any other possible response.status code that we want to handle here? or that's for later?